testing/citest/tests/azure_smoke_test.py

```diff
@@ -0,0 +1,235 @@
+    # pylint: disable=invalid-name
+    self.TEST_APP = bindings['TEST_APP']+AzureSmokeTestScenario.DEFAULT_TEST_ID
+    self.TEST_STACK = bindings['TEST_STACK']+AzureSmokeTestScenario.DEFAULT_TEST_ID
+    self.TEST_SECURITY_GROUP = 'sec_grp_'+ self.TEST_APP
+    self.TEST_SECURITY_GROUP_RG = self.TEST_APP+ '-' + self.bindings['TEST_AZURE_RG_LOCATION']
+
+  #TODO: Perform the login to Azure with the SPN passed as parameters
+  def create_app(self):
+    """Creates OperationContract that creates a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_create_app_operation(
+            bindings=self.bindings, application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_AZURE_ACCOUNT']),
+        contract=contract)
+
+  def delete_app(self):
+    """Creates OperationContract that deletes a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_delete_app_operation(
+            application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_AZURE_ACCOUNT']),
+        contract=contract)
+
+  def create_a_security_group(self):
+    """Creates AzContract for createServerGroup.
+
+    To verify the operation, we just check that the spinnaker security group
+    for the given application was created.
+    This will create a Network Security Group in a Resource Group on your azure Subscription
+    """
+    rules = [
+        {
+            "access": "Allow",
+            "destinationAddressPrefix": "*",
+            "destinationPortRange": "80-80",
+            "direction": "InBound",
+            "endPort": 80,
+            "name": self.TEST_SECURITY_GROUP,
+            "priority": 100,
+            "protocol": "tcp",
+            "sourceAddressPrefix": "*",
+            "sourcePortRange": "*",
+            "startPort": 80
+        }]
+    job = [{
+        "provider": "azure",
+        "application": self.TEST_APP,
+        "appName": self.TEST_APP,
+        "region": self.bindings['TEST_AZURE_RG_LOCATION'],
+        "stack": self.TEST_STACK,
+        "description": "Test - create security group for {app}".format(
+            app=self.TEST_APP),
+        "detail": "",
+        "credentials": self.bindings['SPINNAKER_AZURE_ACCOUNT'],
+        "securityRules": rules,
+        "name": self.TEST_SECURITY_GROUP,
+        "securityGroupName": self.TEST_SECURITY_GROUP,
+        "cloudProvider": "azure",
+        "type": "upsertSecurityGroup",
+        "user": "[anonymous]"
+    }]
+    builder = az.AzContractBuilder(self.az_observer)
+    (builder.new_clause_builder(
+        'Security Group Created', retryable_for_secs=30)
+     .collect_resources(
+         az_resource='network',
+         command='nsg',
+         args=['show', '--name', self.TEST_SECURITY_GROUP, '--resource-group', self.TEST_SECURITY_GROUP_RG],
+         no_resources_ok=True)
+     .contains_pred_list([
```

Comment:
we plan to make the check more comprehensive after we get the initial set of test done.